### PR TITLE
Finally done with GraphML

### PR DESCRIPTION
**Pull request by ebachoura**
This pull request consists of making sure that all of the errors caught by the XML2JS parser were properly converted into GRNsight errors. It also contains tests that covered all of these new errors.

Pull Request Checklist
- [x] Travis C.I. Build Passes
- [x] All Loaded Demos look as Expected
- [x] Reload works as expected
- [x] Loading File with Error brings up error modal, and it looks as expected
- [x] Loading File with Warning brings up warnings modal
- [x] Viewport Option Changing works as expected
- [x] Viewport Size Changing works as expected
- [x] Hide/Show Weights works as expected
- [x] Force Sliders change the number above them and have an effect on the graph
- [x] Locking/Unlocking/Resetting/Undoing the force graph parameters works as expected
- [x] Weighted graph loaded with the "default to black edges" option checked appears as expected
- [x] Import works for both SIF and GraphML
- [x] Graph can be exported to SIF and GraphML

### CODE REVIEW

File: server/controllers/graphml-constants.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    "Invalid characters in closing tag": graphmlErrors.GRAPHML_INCOMPLETE_CLOSING_TAG,
+    "Unquoted attribute value": graphmlErrors.GRAPHML_MISSING_OPEN_QUOTE,
+};
+
+var pairError = function (error) {
+    if (!errorMessageToGraphmlError.hasOwnProperty(error.error)) {
+        return graphmlErrors.GRAPHML_GENERAL_SYNTAX_ERROR(error);
+    }
+    return errorMessageToGraphmlError[error.error](error);
+};
```
Comment (kdahlquist) [inline server/controllers/graphml-constants.js]: Consider rewriting:
```javascript
    return errorMessageToGraphmlError.hasOwnProperty(error.error) ?
        errorMessageToGraphmlError[error.error](error) :
        graphmlErrors.GRAPHML_GENERAL_SYNTAX_ERROR(error);
```

File: server/controllers/importers/graphml.js [hunk 2]
```diff
@@ -50,36 +17,25 @@ module.exports = function (graphml) {
...
-
-        // switch(parseError) {
-        //   case "Invalid attribute name":
-        //     network.errors.push(graphmlErrors.UNKNOWN_ERROR);
-        //     break;
-        //   default:
-        //     network.errors.push(graphmlErrors.GRAPHML_GENERAL_SYNTAX_ERROR);
-        // }
+        var parsedError = parseErr(err);
+        network.errors.push(graphmlConstants.pairError(parsedError));
```
Comment (kdahlquist) [inline server/controllers/importers/graphml.js]: Variable is declared and used only once; consider rewriting as:
```javascript
network.errors.push(graphmlConstants.pairError(parseErr(err)));
```

File: test/import-graphml-tests.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
 /* eslint-disable max-len */
 var expect = require("chai").expect;
-// var extend = require("jquery-extend");
 var fs = require("fs");
 var UTF8 = { encoding: "utf-8" };
 
 var importController = require(__dirname + "/../server/controllers" + "/import-controller")();
 var constants = require(__dirname + "/../server/controllers" + "/constants");
+// var graphmlConstants = require(__dirname + "/../server/controllers" + "/graphml-constants");
```
Comment (kdahlquist) [inline test/import-graphml-tests.js]: Commented-out code; delete.

File: test/import-graphml-tests.js [hunk 4]
```diff
@@ -268,6 +268,293 @@ var misspelledGraphTagTestGraphMl = [
...
+        { source: 2, target: 0, value: 3 },
+        { source: 2, target: 2, value: 0.5 }
+    ],
+
+    errors: [],
+    warnings: [],
+    positiveWeights: [],
+    negativeWeights: [],
+    sheetType: "weighted"
+};
```
Comment (kdahlquist) [inline test/import-graphml-tests.js]: Skip a line between large variable blocks. (do everywhere else)

**Comment by yshin4** [comment]
@dondi we set the timeout to 20000 but the travis still failed. What did we do to fix it last time??

**Comment by yshin4** [comment]
All the other options are checked, we're just waiting on the timeout issue right now.

**Comment by yshin4** [comment]
@ebachoura I figured it out, it was timing out in another file (warning) just like the error file. I increased the timeout so it should work now.

**Pull request merged**